apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +127,57 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp process_package(package, shipping_method_id) do
+    shipping_method =
+      Enum.find(package.shipping_methods, fn %{id: id} ->
+        id == shipping_method_id
+      end)
+
+    package_total =
+      Enum.reduce(
+        [shipping_method.cost],
+        &Money.add!/2
+      )
+
+    Package.update(package, %{
+      cost: shipping_method.cost,
+      total: package_total,
+      shipping_method: shipping_method
+    })
+  end
+
+  @doc """
+  Persists shipping_method_id to packages
+
+  Calculate pacakge total cost Sum(shipping_cost, adjustment_total,promo_total, shipping_cost)
+
+  Update package cost total in DB
+  """
+
+  @spec save_packages_methods(Context.t()) :: Context.t()
+  def save_packages_methods(
+        %Context{valid?: true, struct: %Order{id: order_id} = order} = context
+      ) do
+    %{state: %{selected_shipping_methods: selected_shipping_methods}, multi: multi} = context
+
+    packages = Repo.all(from(p in PackageSchema, where: p.order_id == ^order_id))
```

Comment:
@kanishkablack can we have a method for this like `order_packages` and extract this from here to `orders` model?

Reply:
Sure